todoman/interactive.py

```diff
@@ -0,0 +1,583 @@
+
+    def __init__(self, parent, callback):
+        '''
+        Create a TodomanDatabasesPage instance.
+
+        Callback is the function to call when the page is closed. This function
+        is responsible for passing information to the underlying TodomanPage.
+        It is usually a method of the underlying TodomanPage.
+
+        (TodomanDatabasesPage, TodomanInteractive, function) -> None
+        '''
+        self.parent = parent
+        buttons = []
+        for database in parent.databases:
+            button = urwid.Button("")
+            urwid.connect_signal(button, 'click', self.close_page, database)
+            button._w = urwid.AttrMap(urwid.SelectableIcon(
+                        ["> ", database.name], 2), None, 'selected')
+            buttons.append(button)
+        self.body = urwid.ListBox(urwid.SimpleFocusListWalker(buttons))
+        super().__init__(parent, callback)
+
+    def close_page(self, button, database):
+        '''
+        Close the current page, returning the selected database to the
+        underlying page.
+
+        (TodomanDatabasesPage, Button, Database) -> None
+        '''
+        self.parent.close_page(self, database = database)
+
+    def keypress(self, size, key):
+        '''
+        Make the different commands in the TodomanDatabasesPage view work.
+
+        (TodomanDatabasesPage, int(?), str) -> str
+        '''
+        if key == 'esc':
+            self.parent.close_page(self)
+            return None
+        if key == 'j':
+            return super().keypress(size, 'down')
+        if key == 'k':
+            return super().keypress(size, 'up')
+        return super().keypress(size, key)
+
+class TodomanInteractive(object):
+    '''
+    Class to run the interactive, curses-based interface to Todoman.
+    '''
+
+    palette = [
+        ('statusbar', 'light gray', 'dark blue'),
+        ('reversed', 'standout', ''),
+        ('error', 'light red', '')
+    ]
+
+    def __init__(self, databases, formatter):
+        '''
+        Create a TodomanInteractive instance based on the Database objects that
+        the regular Todoman cli module passes.
+
+        (TodomanInteractive, [Database], TodoFormatter) -> None
+        '''
+        self.databases = list(databases)
+        self.databases.sort(key = lambda db: db.name)
+        self.formatter = formatter
+        top = urwid.Filler(urwid.Text("Press q to exit"))
+        self.loop = urwid.MainLoop(top,
+                                   self.palette,
+                                   unhandled_input = self.unhandled_input)
+        firstPage = TodomanItemListPage(self, None, self.databases[0])
+        self.pageCounter = 0 
+        self._open_page(firstPage)
+
+        self.loop.run()
+
+    def unhandled_input(self, key):
+        '''
+        Handles all the key presses that are application-wide.
+
+        (TodomanInteractive, str) -> None
+        '''
+        if key in ('q', 'Q'):
+            raise urwid.ExitMainLoop()
```

Comment:
When editing an entry, if a checkbox if focus and I press <kbd>q</kbd>, it immediately exists, without any confirmation, etc.